alerting/temporal/src/index.ts

```diff
@@ -0,0 +1,320 @@
+): Promise<MetricData> => {
+  const response = await axios.get(PROM_QUERY_URL, {
+    httpsAgent,
+    params: {
+      ...basePrometheusQueryParams,
+      query: `rate(${metricName}[1m])`,
+      start: queryWindow.startSecondsSinceEpoch.toFixed(0),
+      end: queryWindow.endSecondsSinceEpoch.toFixed(0),
+    },
+  });
+
+  return queryResponseDataSchema.parse(response.data).data;
+};
+
+const convertPrometheusCountToDatadogRateSeries = (
+  metricName: string,
+  metricData: MetricData
+): v2.MetricSeries[] =>
+  metricData.result.map((prometheusMetric) => ({
+    // We need to inform datadog of the interval for this metric
+    interval: PROMETHEUS_STEP_SECONDS,
+    // Make it easier for the datadog user to understand what this metric is
+    metric: DATADOG_METRIC_PREFIX + metricName.split("_count")[0] + "_rate1m",
+    // Type 2 is a "rate" metric
+    type: 2,
+    points: prometheusMetric.values.map(([timestamp, value]) => {
+      return {
+        timestamp: timestamp,
+        value: parseFloat(value),
+      };
+    }),
+    tags: Object.entries(prometheusMetric.metric)
+      .filter(([key]) => key !== "__rollup__")
+      // Datadog tags can't be longer than 200 characters
+      .map(([key, value]) => `${key}:${value.substring(0, 200)}`),
+  }));
+
+const queryPrometheusHistogram = async (
+  metricName: string,
+  quantile: number,
+  queryWindow: QueryWindow
+): Promise<MetricData> => {
+  const response = await axios.get(PROM_QUERY_URL, {
+    httpsAgent,
+    params: {
+      ...basePrometheusQueryParams,
+      query: `histogram_quantile(${quantile}, sum(rate(${metricName}[1m])) by (temporal_account,temporal_namespace,operation,le))`,
+      start: queryWindow.startSecondsSinceEpoch.toFixed(0),
+      end: queryWindow.endSecondsSinceEpoch.toFixed(0),
+    },
+  });
+
+  return queryResponseDataSchema.parse(response.data).data;
+};
+
+const convertPrometheusHistogramToDatadogGuageSeries = (
+  metricName: string,
+  quantile: number,
+  metricData: MetricData
+): v2.MetricSeries[] =>
+  metricData.result.map((prometheusMetric) => ({
+    // Make it easier for the datadog user to understand what this metric is
+    metric:
+      DATADOG_METRIC_PREFIX +
+      metricName.split("_bucket")[0] +
+      "_P" +
+      quantile * 100,
+    // Type 2 is a "guage" metric
+    type: 3,
+    points: prometheusMetric.values.map(([timestamp, value]) => {
+      return {
+        timestamp: timestamp,
+        value: parseFloat(value),
+      };
+    }),
+    tags: Object.entries(prometheusMetric.metric)
+      .filter(([key]) => key !== "__rollup__")
+      // Datadog tags can't be longer than 200 characters
+      .map(([key, value]) => `${key}:${value.substring(0, 200)}`),
+  }));
+
+const main = async () => {
+  const { countMetricNames, histogramMetricNames } = await getMetricNames();
+
+  console.log({
+    level: "info",
+    message: "Polling metrics",
+    countMetricNames,
+    histogramMetricNames,
+  });
+
+  while (true) {
```

Comment:
How do we escape when the machine is supposed to rollout? Does it just get killed?

Reply:
Unless it's catching SIGTERMs it will get kiled